### ISSUE TITLE
Write FIT file/stream

### ISSUE THREAD

**Issue by DevronB** [sports-alliance/sports-lib#68]
Question:
Hi, is it possible to convert TCX/GPS to FIT? or even just the ability to create a FIT file/stream with this? I've been looking for a JS library to create FIT files. If not, any chance this might be added in future?

Cheers

**Comment by jimmykane** [comment]
Unfortunately its quite a task todo this. Currently I also dev the https://github.com/jimmykane/fit-parser which parses the FIT files. So far I have not seen any JS lib doing the encoding. I would love that but all people (including me) that have tried have "failed" due to the "complexity " I suppose.

**Comment by jastax** [comment]
I'd love to see this too :)

**Comment by mrloop** [comment]
I'm interested too. Saw this https://developer.garmin.com/fit/file-types/activity/ . @jimmykane would you be willing to share your attempt at fit encoding?

**Comment by mrloop** [comment]
Further reading and this is the protocol https://developer.garmin.com/fit/protocol/, more than a weekends work I think.

**Comment by mrloop** [comment]
I tried using https://github.com/opal/opal to convert a ruby lib for encoding / decoding fit files to javascript, https://github.com/jpablobr/f4r/blob/main/lib/f4r.rb. But the outputted code was hard to read, and I configured it to ignore missing require statements, that is not convert all the required libs as well, so will not run. I wonder if porting this library by hand, or another to javascript would be the easist approach.
